### PR TITLE
MRG: Better shape checks for SourceEstimate

### PR DESCRIPTION
**Pull request by larsoner**
I've run into tricky problems accidentally passing vector data `(n, 3, t)` to SourceEstimate. This implements `data` size checks for our classes.

### CODE REVIEW

**Comment by larsoner** [comment]
Argh this exposes a problem where `VolSourceEstimate` does not have a `Vector` type, but should. I guess I'll add it. We are getting quite a few `*SourceEstimate` types.

File: mne/tests/test_source_estimate.py [hunk 1]
```diff
@@ -804,24 +804,30 @@ def test_mixed_stc():
     assert isinstance(stc_out, MixedSourceEstimate)
 
 
-def test_vec_stc():
-    """Test vector source estimate."""
+@pytest.mark.parametrize('class_',
+                         (VectorSourceEstimate, VolVectorSourceEstimate))
+def test_vec_stc(class_):
```
Comment (agramfort) [inline mne/tests/test_source_estimate.py]: can you call class_ klass to be consistent in naming elsewhere?

File: mne/source_estimate.py [hunk 2]
```diff
@@ -407,10 +408,16 @@ def _make_stc(data, vertices, src_type=None, tmin=None, tstep=None,
     elif src_type in ('volume', 'discrete'):
         if vector:
             data = data.reshape((-1, 3, data.shape[-1]))
-        stc = VolSourceEstimate(data, vertices=vertices, tmin=tmin,
-                                tstep=tstep, subject=subject)
+            class_ = VolVectorSourceEstimate
```
Comment (agramfort) [inline mne/source_estimate.py]: class_ -> klass

File: mne/source_estimate.py [hunk 3]
```diff
@@ -529,6 +536,48 @@ def __init__(self, data, vertices=None, tmin=None, tstep=None,
...
+        s += ", tstep : %s (ms)" % (1e3 * self.tstep)
+        s += ", data shape : %s" % (self.shape,)
+        return "<%s  |  %s>" % (type(self).__name__, s)
+
+    @property
+    def _vertices_list(self):
+        return self.vertices
+
+    @verbose
+    def save(self, fname, ftype='h5', verbose=None):
```
Comment (agramfort) [inline mne/source_estimate.py]: does it mean we write to hdf5 by default also for surface SourceEstimates with this?

Comment (larsoner) [reply mne/source_estimate.py]: No, `SourceEstimate` overrides this and only calls `super` if they request `h5`

**Comment by agramfort** [comment]
ok ! LGTM then

>

**Comment by larsoner** [comment]
Renaming complete

**Comment by agramfort** [comment]
thx @larsoner

**Pull request merged**